src/CommunityToolkit.Maui/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,171 @@
+namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// Extension methods for registering Views and ViewModels in <see cref="IServiceCollection"/>
+/// </summary>
+public static class ServiceCollectionExtensions
+{
+	/// <summary>
+	/// Adds a <see cref="BindableObject"/> of the type specified in <typeparamref name="TView"/> and a ViewModel
+	/// of the type specified in <typeparamref name="TViewModel"/> to the specified
+	/// <see cref="IServiceCollection"/> with <see cref="ServiceLifetime.Transient"/> lifetime.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="BindableObject"/></typeparam>
+	/// <typeparam name="TViewModel">The type of the ViewModel to add. Constrained to 
+	/// <see cref="INotifyPropertyChanged"/></typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+	/// <returns>A reference to this instance after the operation has completed.</returns>
+	/// <remarks>Developers are still responsible for assigning the injected instance of 
+	/// <typeparamref name="TViewModel" />
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddTransient<TView, TViewModel>(this IServiceCollection services)
+		where TView : BindableObject
+		where TViewModel : class, INotifyPropertyChanged
+	{
+		return services
+			.AddTransient<TViewModel>()
+			.AddTransient<TView>();
+	}
+
+	/// <summary>
+	/// Adds a <see cref="NavigableElement"/> of the type specified in <typeparamref name="TView"/> and a ViewModel
+	/// of the type specified in <typeparamref name="TViewModel"/> to the specified
+	/// <see cref="IServiceCollection"/> with <see cref="ServiceLifetime.Transient"/> lifetime
+	/// and registers a MAUI Shell route in <see cref="Routing"/> using the value of 
+	/// <paramref name="route"/> as the route.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="NavigableElement"/></typeparam>
+	/// <typeparam name="TViewModel">The type of the ViewModel to add. Constrained to 
+	/// <see cref="INotifyPropertyChanged"/></typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+	/// <param name="route">Route at which this page will be registered within Shell routing.</param>
+	/// <param name="factory">RouteFactory to be used while creating the <see cref="NavigableElement"/> 
+	/// for the route. Defaults to TypeRouteFactory.</param>
+	/// <returns>A reference to this instance after the operation has completed.</returns>
+	/// <remarks>Developers are still responsible for assigning the injected instance of <typeparamref name="TViewModel" /> 
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddTransientWithShellRoute<TView, TViewModel>(this IServiceCollection services, string route, RouteFactory? factory = null)
```

Comment:
What do you think if we simplify the name to AddTransient?

Reply:
I think it makes sense to keep the name `AddTransientWithShellRoute` since we're accepting parameters `string route` and `RouteFactory? factory`.

If we shorten it to `AddTransient<TView, TViewModel>(this IServiceCollection services, string route, RouteFactory? factory = null)`, it may be confusing for devs who see this appear in intellisence. And then we're relying on developers to read the docs which, as we've learned, is never something we want to rely on 😆